### DISCUSSION TITLE
How to associate Firefox Container with specific bookmarks?

### DISCUSSION THREAD

**Discussion by janban-outlook** [mozilla/multi-account-containers#2685] [Q&A] [answered]
I'm using Firefox's Multi-Account Containers extension to manage multiple Gmail accounts. Currently, my workflow is:

Open a new container tab (e.g., "Work" container)
Click the Gmail bookmark
Repeat with different container for personal Gmail

Is there a way to create bookmarks that automatically open in specific containers? For example, I'd like to have:

"Work Gmail" bookmark that opens Gmail in my Work container
"Personal Gmail" bookmark that opens Gmail in my Personal container

This would streamline the process by eliminating the manual container selection step.

OS: Arch Linux
Firefox version: 131.0.3
Multi-Account Containers extension version: 8.2.0

**Comment by tabasc** [comment]
Yeah, without either this or the ability to set full URLs the add on is next to useless.

If you can only set the domain, might as well not even bother using containers at all.

Pretty hilariously basic design flaw.

**Comment by kelimuttu** [comment]
The problem with this is that the URL will be the same whether it's for work Gmail or personal Gmail, so I personally think there's no easy way to work around this.

**Reply by janban-outlook** [reply]
Makes sense. But when you look at the current workflow, it is two clicks to go from the bookmark to the website you want to see.
Wouldn't it be possible to extend the bookmark definition with a container name property. That would solve the issue.

![image](https://github.com/user-attachments/assets/346a80b0-5aac-4c2f-945b-1f31f740cdce)

**Reply by kelimuttu** [reply, marked as answer]
Oh, I see.. So the integration is more on the bookmark level rather than the URL. I found it similar to [this one](https://github.com/mozilla/multi-account-containers/issues/1050). If you don't think they're the same, feel free to open a new issue so we can document it as a feature request.

**Comment by tabasc** [comment]
Yes for generic account pages the full URL would not work.

My intended use case was to open specific trello boards in specific accounts, so for that it would work as each have a unique URL.

**Comment by janban-outlook** [comment]
Your answer made me searching for a way to get a unique url for Gmail per account. And I found it [here](https://webapps.stackexchange.com/a/97952). Now I have two bookmarks for my private and work email. Super!
Hope this keeps working...

UPDATE: Unfortunately this doesn't work when you are not logged in first :cry:

**Reply by tabasc** [reply]
That might be specific to Gmail. What I've found is that as long as you have cookies enabled and have auto login for each account, it should auto log in to the "correct" account when opened in that container. I use it this way with a couple of MS accounts for services which rely on MS login, however these services don't share the same domain name. (I put correct in quotes because it's actually whichever account was last left logged in in that container)
